### ISSUE TITLE
Array required field validations are not working when object item not mentioned in requireds 

### ISSUE THREAD

**Issue by varund29** [hamzahamidi/ajsf#214]
**Array required field validations are not working when object item not mentioned in required**
which template: below all temples having this issue.

* [x] MaterialDesignFrameworkModule — Material Design
* [x] Bootstrap3FrameworkModule — Bootstrap 3
* [x] Bootstrap4FrameworkModule — Bootstrap 4
* [x] NoFrameworkModule — plain HTML
* [ ] Other (please specify below)
  
I have a schema, which contains array items(item types->string,number,object). required field validations are only working fine in combination of primitive types(["Name","ID"]) + object type(["Connection"])  as below

"required": ["Name","ID","Connection"] 
here 
"Name","ID" are type string
"Connection" is type object

till no issues.

The issue is when i use only primitive types(["Name","ID"]) in required fields validations are not working(without object type i.e "Connection") as below
"required": ["Name","ID"] 

**To Reproduce**

1. here is  [my schema](https://github.com/varund29/json-schema/blob/master/json-schem.json) 
using [this tool](https://angular2-json-schema-form.firebaseapp.com/)  >>  in the Form Output section below are the validation i'm getting.
Actual output ( "required": ["Name","ID","Connection"]  for "Native" node)

Properties: should have required property 'Configuration Name'
Properties: should have required property 'Configuration Description'
Properties: should have required property 'Configuration Version'
Components.Native[0]: should have required property 'ID'
Components.Native[0]: should have required property 'Name'

**Expected behavior**
I'm expecting below output in bold( "required": ["Name","ID"]  for "IP" node), Because in "IP"  required field i didn't mentioned object type("Connection"), that is why validations are not working for "IP". 
Properties: should have required property 'Configuration Name'
Properties: should have required property 'Configuration Description'
Properties: should have required property 'Configuration Version'
Components.Native[0]: should have required property 'ID'
Components.Native[0]: should have required property 'Name'
**Components.IP[0]: should have required property 'ID'
Components.IP[0]: should have required property 'Name'**

In summary:
below validations are not coming 
**Components.IP[0]: should have required property 'ID'
Components.IP[0]: should have required property 'Name'`**

- OS: Windows
- Browser : Chrome

**Comment by varund29** [comment]
Can i get any update on this issue please..

**Comment by govindthakur25** [comment]
Did you try putting the required object after the properties?

A [gist](https://gist.github.com/govindthakur25/1e2cf6b7c77e9583c023342e462596ea) for your reference.

**Comment by varund29** [comment]
@govindthakur25 yes i used "required" after properties. Have you tried my scheme give above([my schema](https://github.com/varund29/json-schema/blob/master/json-schem.json))
